pkg/dockerregistry/server/auth.go

```diff
@@ -125,12 +139,15 @@ var (
 
 func newAccessController(options map[string]interface{}) (registryauth.AccessController, error) {
 	log.Info("Using Origin Auth handler")
-	realm, ok := options["realm"].(string)
+	realm, ok := options[RealmKey].(string)
 	if !ok {
 		// Default to openshift if not present
 		realm = "origin"
 	}
-	return &AccessController{realm: realm, config: DefaultRegistryClient.SafeClientConfig()}, nil
+
+	tokenRealm, _ := options[TokenRealmKey].(string)
```

Comment:
the auth component will answer with bearer token challenges if it has a configured token realm. if it doesn't `tokenRealm` will be an empty string and the auth component will just return the unauthorized challenge it does today